### ISSUE TITLE
Error: Could not locate the bindings file (wrangler)

### ISSUE THREAD

**Issue by gkiely** [oven-sh/bun#3207]
### What platform is your computer?

Darwin 21.3.0 arm64 arm

### How did you attempt to install or upgrade?

bun i wrangler
wrangler

> miniflare is using better-sqlite3
```
    "node_modules/miniflare": {
      "version": "3.0.0",
      "resolved": "https://registry.npmjs.org/miniflare/-/miniflare-3.0.0.tgz",
      "integrity": "sha512-CW8yS00pQCbq2o4K8drePzJBkE0heL9cCY/O8DcYJOHd4M0RVutjrp+LqUP4hfzweZ+LqrLo8+Rzy4cORisQoQ==",
      "dev": true,
      "dependencies": {
        "better-sqlite3": "^8.1.0",
```

### What do you see instead?

[mf:err] Error: Could not locate the bindings file. Tried:
 → /Users/grantkiely/code/my-project/node_modules/better-sqlite3/build/better_sqlite3.node
 → /Users/grantkiely/code/my-project/node_modules/better-sqlite3/build/Debug/better_sqlite3.node
 → /Users/grantkiely/code/my-project/node_modules/better-sqlite3/build/Release/better_sqlite3.node
 → /Users/grantkiely/code/my-project/node_modules/better-sqlite3/out/Debug/better_sqlite3.node
 → /Users/grantkiely/code/my-project/node_modules/better-sqlite3/Debug/better_sqlite3.node
 → /Users/grantkiely/code/my-project/node_modules/better-sqlite3/out/Release/better_sqlite3.node
 → /Users/grantkiely/code/my-project/node_modules/better-sqlite3/Release/better_sqlite3.node
 → /Users/grantkiely/code/my-project/node_modules/better-sqlite3/build/default/better_sqlite3.node
 → /Users/grantkiely/code/my-project/node_modules/better-sqlite3/compiled/20.2.0/darwin/arm64/better_sqlite3.node
 → /Users/grantkiely/code/my-project/node_modules/better-sqlite3/addon-build/release/install-root/better_sqlite3.node
 → /Users/grantkiely/code/my-project/node_modules/better-sqlite3/addon-build/debug/install-root/better_sqlite3.node
 → /Users/grantkiely/code/my-project/node_modules/better-sqlite3/addon-build/default/install-root/better_sqlite3.node
 → /Users/grantkiely/code/my-project/node_modules/better-sqlite3/lib/binding/node-v115-darwin-arm64/better_sqlite3.node
    at bindings (/Users/grantkiely/code/my-project/node_modules/bindings/bindings.js:126:9)
    at new Database (/Users/grantkiely/code/my-project/node_modules/better-sqlite3/lib/database.js:48:64)
    at createFileStorage (/Users/grantkiely/code/my-project/node_modules/miniflare/dist/src/index.js:3627:14)
    at GatewayFactory.getStorage (/Users/grantkiely/code/my-project/node_modules/miniflare/dist/src/index.js:3686:12)
    at GatewayFactory.get (/Users/grantkiely/code/my-project/node_modules/miniflare/dist/src/index.js:3692:26)
    at CacheRouter.put (/Users/grantkiely/code/my-project/node_modules/miniflare/dist/src/index.js:4880:41)
    at CacheRouter.route (/Users/grantkiely/code/my-project/node_modules/miniflare/dist/src/index.js:3822:27)
    at #handleLoopbackPlugins (/Users/grantkiely/code/my-project/node_modules/miniflare/dist/src/index.js:8919:54)
    at #handleLoopback (/Users/grantkiely/code/my-project/node_modules/miniflare/dist/src/index.js:8988:53)
    at Server.emit (node:events:523:35)

### Additional information

Related: https://github.com/oven-sh/bun/issues/2130

If you need a repro I can set one up.

**Comment by Jarred-Sumner** [comment]
This will be fixed by #2073 which @alexlamsl is working on

**Comment by gkiely** [comment]
Cool I'll close and retest after that ticket is done.

**Comment by gkiely** [comment]
@Jarred-Sumner Just a heads up I'm still seeing this error in the canary build. I can setup a minimal repro if that would be helpful.

**Comment by urgobalt** [comment]
I have encountered the same error, and the fix I have found is to go into `better-sqlite3` and manually run the node script called `install`. Other package managers, such as yarn, seem to run this script during the installation phase, while bun does not.

**Comment by gkiely** [comment]
> I have encountered the same error, and the fix I have found is to go into `better-sqlite3` and manually run the node script called `install`. Other package managers, such as yarn, seem to run this script during the installation phase, while bun does not.

I couldn't get that working with:
```
"wrangler": "^3.7.0",
"better-sqlite3": "^8.6.0",
```

**Comment by ukd1** [comment]
I'm also hitting this issue

**Comment by Caleb-Irwin** [comment]
Running `bun install` and then running `pnpm i better-sqlite3` (or `npm i better-sqlite3`) fixed my `Error: Could not locate the bindings file.` error. _Everything_ else works great with bun.

**Comment by essential-randomness** [comment]
Likewise, with the latest `better-sqlite3` (8.6.0) but no `wrangler`.

**Comment by bndkt** [comment]
For me this issue is resolved when adding

```package.json
"trustedDependencies": [
    "better-sqlite3"
  ]
```

to package.json and re-installing dependencies. Notice that you need to add **better-sqlite3**, not wrangler to trustedDependencies.

**Comment by essential-randomness** [comment]
> For me this issue is resolved when adding
> 
> ```json
> "trustedDependencies": [
>     "better-sqlite3"
>   ]
> ```
> 
> to package.json and re-installing dependencies. Notice that you need to add **better-sqlite3**, not wrangler to trustedDependencies.

I tried this, but also had to `bun add node-gyp` afterwards to solve a "No prebuilt binaries found (target=19.9.0 runtime=node arch=x64 libc= platform=darwin)" error. Then better-sqlite3 did run its own installer during `bun install`.

Unfortunately, once I go run the code i get:
```
$ bun run --watch src/index.ts
dyld[92161]: missing symbol called
```

**Comment by ukd1** [comment]
> I'm also hitting this issue

cloudflare/workers-sdk#3964 - I figured out what it was for me; a simple nuking and reinstalling fixed using `npm install`

**Comment by gkiely** [comment]
This has been resolved for me as of `1.0.15`
